src/components/ColorModeContext.tsx

```diff
@@ -0,0 +1,77 @@
+import * as React from "react";
+import {createContext, useContext, useMemo, useState, ReactNode} from "react";
+import {ThemeProvider, createTheme, PaletteMode} from "@mui/material";
+import {purple, teal, grey} from "@mui/material/colors";
+import {theme as baseTheme} from "../utils/theme";
+
+interface ColorModeContextType {
+    toggleColorMode: () => void;
+}
+
+const ColorModeContext = createContext<ColorModeContextType | undefined>(
+    undefined
+);
+
+// eslint-disable-next-line react-refresh/only-export-components
+export const useColorMode = () => {
+    const context = useContext(ColorModeContext);
+    if (!context) {
+        throw new Error("useColorMode must be used within a ColorModeProvider");
+    }
+    return context;
+};
+
+const getDesignTokens = (mode: PaletteMode) => ({
```

Comment:
can we move this part with colors to the theme file? It`s convenient to see all your colors there